presto-geospatial-toolkit/src/main/java/com/facebook/presto/geospatial/JtsGeometrySerde.java

```diff
@@ -32,319 +32,206 @@
-     */
-    public static Slice serialize(Geometry geometry)
+    private static Geometry readGeometry(BasicSliceInput input, GeometryType type)
     {
-        SliceOutput output = new DynamicSliceOutput(100);
-
-        boolean inGeometryCollection = geometry.getGeometryType().equals(JTS_GEOMETRY_COLLECTION);
-        if (inGeometryCollection) {
-            output.writeByte(GeometryType.GEOMETRY_COLLECTION.code());
-            for (int i = 0; i < geometry.getNumGeometries(); i++) {
-                writeGeometry(geometry.getGeometryN(i), output, true);
-            }
-        }
-        else {
-            writeGeometry(geometry, output, false);
-        }
-        return output.slice();
-    }
-
-    private static void writeGeometry(Geometry geometry, SliceOutput output, boolean inGeometryCollection)
-    {
-        switch (geometry.getGeometryType()) {
-            case JTS_POINT:
-                writePoint((Point) geometry, output, inGeometryCollection);
-                break;
-            case JTS_MULTI_POINT:
-                writeMultiPoint((MultiPoint) geometry, output, inGeometryCollection);
-                break;
-            case JTS_LINESTRING:
-            case JTS_MULTI_LINESTRING:
-                writePolyline(geometry, output, inGeometryCollection);
-                break;
-            case JTS_POLYGON:
-            case JTS_MULTI_POLYGON:
-                writePolygon(geometry, output, inGeometryCollection);
-                break;
-            default:
-                throw new IllegalArgumentException("Unsupported geometry type : " + geometry.getGeometryType());
-        }
-    }
-
-    private static Geometry readGeometry(SliceInput input)
-    {
-        requireNonNull(input, "input is null");
-        int geometryType = input.readInt();
-        switch (EsriShapeType.valueOf(geometryType)) {
+        switch (type) {
             case POINT:
-                return readPoint(input);
-            case POLYLINE:
-                return readPolyline(input);
-            case POLYGON:
-                return readPolygon(input);
+                return readPoint(input, true);
             case MULTI_POINT:
                 return readMultiPoint(input);
+            case LINE_STRING:
+                return readPolyline(input, false);
+            case MULTI_LINE_STRING:
+                return readPolyline(input, true);
+            case POLYGON:
+                return readPolygon(input, false);
+            case MULTI_POLYGON:
+                return readPolygon(input, true);
+            case GEOMETRY_COLLECTION:
+                return readGeometryCollection(input);
             default:
-                throw new UnsupportedOperationException("Invalid geometry type: " + geometryType);
+                throw new UnsupportedOperationException("Unexpected type: " + type);
         }
     }
 
-    private static Point readPoint(SliceInput input)
+    private static Point readPoint(SliceInput input, boolean skipType)
```

Comment:
`skipType` parameter is confusing me; since there are only 2 calls to `readPoint` - one with `skipType = true` and one with `skipType = false` it would be better to move `skipEsriType(input);` to the caller

Reply:
It was just for a transition. I removed this parameter in `Custom point serialization` commit anyhow.